docs/guide/obs-integrations.asciidoc

```diff
@@ -0,0 +1,185 @@
+guide will walk you through the setup process.
+
+Because logging frameworks and formats vary greatly between different programming languages,
+there is no one-size-fits-all approach for ingesting your logs into Elasticsearch.
+The following tips should hopefully get you going in the right direction:
+
+**Download Filebeat**
+
+There are many ways to download and get started with Filebeat.
+Read the {filebeat-ref}/filebeat-installation.html[Filebeat Installation] guide to determine which is best for you.
+
+**Configure Filebeat**
+
+Modify the {filebeat-ref}/filebeat-configuration.html[`filebeat.yml`] configuration file to your needs.
+Here are some recommendations:
+
+* Set `filebeat.inputs` to point to the source of your logs
+* Point Filebeat to the same Elastic Stack that is receiving your APM data
+  * If you're using Elastic cloud, set `cloud.id` and `cloud.auth`.
+  * If your using a manual setup, use `output.elasticsearch.hosts`.
+
+[source,yml]
+----
+filebeat.inputs:
+- type: log <1>
+  paths: <2>
+    - /var/log/*.log
+cloud.id: "staging:dXMtZWFzdC0xLmF3cy5mb3VuZC5pbyRjZWMNjN2Q3YTllOTYyNTc0Mw==" <3>
+cloud.auth: "elastic:YOUR_PASSWORD" <4>
+----
+<1> Configures the `log` input
+<2> Path(s) that must be crawled to fetch the log lines
+<3> Used to resolve the Elasticsearch and Kibana URLs for Elastic Cloud
+<4> Authorization token for Elastic Cloud
+
+**JSON logs**
+
+For JSON logs you can use the {filebeat-ref}/filebeat-input-log.html[`log` input] to read lines from log files.
+Here's what a sample configuration might look like:
+
+[source,yml]
+----
+filebeat.inputs:
+  json.keys_under_root: true <1>
+  json.add_error_key: true <2>
+  json.message_key: message <3>
+----
+<1> `true` copies JSON keys to the top level in the output document
+<2> Tells Filebeat to add an `error.message` and `error.type: json` key in case of JSON unmarshalling errors
+<3> Specifies the JSON key on which to apply line filtering and multiline settings
+
+// For JSON logs you'll need to use the {filebeat-ref}/decode-json-fields.html[decode_json_fields] processor.
+//
+// [source,yml]
+// ----
+// processors:
+// - decode_json_fields:
+//     fields: ["message"] <1>
+//     target: "" <2>
+//     overwrite_keys: true <3>
+// ----
+// <1> The fields containing JSON strings to decode
+// <2> The field under which the decoded JSON will be written. `""` merges the decoded JSON
+// fields into the root of the event.
+// <3> Specifies whether the keys that already exist in the event are overwritten.
+
+**Parsing unstructured logs**
+
+Consider the following log that is decorated with the `transaction.id` and `trace.id` fields:
+
+[source,log]
+----
+2019-09-18 21:29:49,525 - django.server - ERROR - "GET / HTTP/1.1" 500 27 | elasticapm transaction.id=fcfbbe447b9b6b5a trace.id=f965f4cc5b59bdc62ae349004eece70c span.id=None
+----
+
+Once ingested into Elasticsearch, you'd be able to use Elasticsearch's {ref}/grok-processor.html[Grok Processor]
```

Comment:
Perhaps link to https://www.elastic.co/guide/en/beats/filebeat/current/configuring-ingest-node.html? This shows how to load the pipeline into Elasticsearch, and how to modify filebeat.yml to use it.

Reply:
Or we could also document how to actually create the ingest pipeline. Similar to https://github.com/elastic/apm-agent-java/issues/168#issuecomment-417268839

Reply:
Yeah, that's probably better. It doesn't take much more to add the "PUT" line, and then we just need to describe adding the pipeline name to elasticsearch.output in filebeat.yml.